src/shared/components/query/QueryStore.ts

```diff
@@ -234,7 +234,7 @@ export class QueryStore
 	@observable private _maxTreeDepth:number = (window as any).maxTreeDepth;
 	@computed get maxTreeDepth()
 	{
-		return this.forDownloadTab ? 1 : this._maxTreeDepth;
+		return (this.forDownloadTab && this._maxTreeDepth > 0) ? 1 : this._maxTreeDepth;

```

Comment:
also, this would be a good spot for a comment.  whenever we implement business logic that requires further knowledge to understand, a comment is called for.